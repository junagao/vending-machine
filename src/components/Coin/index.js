export { default as CoinItem } from './CoinItem';
export { default as CoinList } from './CoinList';
export { default as CoinSlot } from './CoinSlot';
export { default as CollectRefund } from './CollectRefund';
export { default as DisplayDepositedAmount } from './DisplayDepositedAmount';
export { default as RefillCoins } from './RefillCoins';
export { default as Wallet } from './Wallet';
