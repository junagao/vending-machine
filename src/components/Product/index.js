export { default as CollectProduct } from './CollectProduct';
export { default as ProductList } from './ProductList';
export { default as ProductItem } from './ProductItem';
export { default as RefillProductStock } from './RefillProductStock';
