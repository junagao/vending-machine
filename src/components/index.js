export { default as CoinSlot } from 'components/CoinSlot';
export { default as CollectProduct } from 'components/CollectProduct';
export { default as VendingMachine } from 'components/VendingMachine';
export { default as VendingMachineScreen } from 'components/VendingMachineScreen';
export { default as Wallet } from 'components/Wallet';
export { default as ProductList } from 'components/ProductList';
export { default as Header } from 'components/Header';
export { default as CollectRefund } from 'components/CollectRefund';
