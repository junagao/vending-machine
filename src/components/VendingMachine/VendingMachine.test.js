import React from 'react';
import { shallow } from 'enzyme';
import { VendingMachine } from 'components';

describe('VendingMachine', () => {
  let wrapper;
  const machineCoinsAmountMock = 10;
  const productsMock = [
    {
      id: 'coke',
      name: 'coke',
      img: 'images/coke.png',
      price: 1,
      currentStock: 10,
    },
    {
      id: 'sprite',
      name: 'sprite',
      img: 'images/sprite.png',
      price: 1.25,
      currentStock: 5,
    },
    {
      id: 'fanta',
      name: 'fanta',
      img: 'images/fanta.png',
      price: 1.25,
      currentStock: 11,
    },
  ];
  const selectedProductNameMock = 'fanta';
  const selectedProductImgMock = 'images/fanta.png';

  beforeEach(() => {
    wrapper = shallow(
      <VendingMachine
        onDrop={jest.fn()}
        onDragOver={jest.fn()}
        onDragEnter={jest.fn()}
        onDragLeave={jest.fn()}
        isDropZone
        machineCoinsAmount={machineCoinsAmountMock}
        products={productsMock}
        onSelectProduct={jest.fn()}
        onCollectCoinRefund={jest.fn()}
        selectedProductName={selectedProductNameMock}
        selectedProductImg={selectedProductImgMock}
        onCollectProduct={jest.fn()}
        isCollected
      />,
    );
  });

  it('should render correctly given the required props', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render one VendingMachineContainer element', () => {
    expect(wrapper.find('VendingMachineContainer').length).toEqual(1);
  });

  it('should render one CoinSlot element', () => {
    expect(wrapper.find('CoinSlot').length).toEqual(1);
  });

  it('should render one DisplayDepositedAmount element', () => {
    expect(wrapper.find('DisplayDepositedAmount').length).toEqual(1);
  });

  it('should render one ProductList element', () => {
    expect(wrapper.find('ProductList').length).toEqual(1);
  });

  it('should render one CollectRefund element', () => {
    expect(wrapper.find('CollectRefund').length).toEqual(1);
  });

  it('should render one CollectProduct element', () => {
    expect(wrapper.find('CollectProduct').length).toEqual(1);
  });
});
