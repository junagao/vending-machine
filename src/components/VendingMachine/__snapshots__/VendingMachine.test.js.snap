// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VendingMachine should render correctly given the required props 1`] = `
<VendingMachineContainer>
  <CoinSlot
    isDropZone={true}
    onDragEnter={[MockFunction]}
    onDragLeave={[MockFunction]}
    onDragOver={[MockFunction]}
    onDrop={[MockFunction]}
  />
  <DisplayDepositedAmount
    machineCoinsAmount={10}
  />
  <ProductList
    onSelectProduct={[MockFunction]}
    products={
      Array [
        Object {
          "currentStock": 10,
          "id": "coke",
          "img": "images/coke.png",
          "name": "coke",
          "price": 1,
        },
        Object {
          "currentStock": 5,
          "id": "sprite",
          "img": "images/sprite.png",
          "name": "sprite",
          "price": 1.25,
        },
        Object {
          "currentStock": 11,
          "id": "fanta",
          "img": "images/fanta.png",
          "name": "fanta",
          "price": 1.25,
        },
      ]
    }
  />
  <CollectRefund
    machineCoinsAmount={10}
    onCollectCoinRefund={[MockFunction]}
  />
  <CollectProduct
    isCollected={true}
    onCollectProduct={[MockFunction]}
    selectedProductImg="images/fanta.png"
    selectedProductName="fanta"
  />
</VendingMachineContainer>
`;
